src/modules/alerting.ts

```diff
@@ -72,29 +109,22 @@ class AlertingApi extends BaseApi {
      * };
      * myClientApp.alerting.showToastPopup("Hello world", "Hello :earth_americas: How are *you* doing today?", options);
      * ```
-     * 
+     *
      * @param title - Toast title.
      * @param message - Toast Message.  Supports emoticons, emoji (unicode, shortcodes) and markdown (with markdwownMessage boolean).
      * @param options - Additonal toast options.
      *
      * @since 1.0.0
      */
-    showToastPopup(title: string, message: string, options: {
-        /**
-         * The id of the message.  Duplicate IDs will replace each other in the toast display.  All IDs will be
-         * namespaced with your app ID to avoid collisions.  (Default will just be your app's namespace and will not support multiple messages)
-         * */
-        id?: string;
-        /** Toast type, valid options are 'error', 'info', 'success'.  (Default is 'info') */
-        type?: 'error' | 'info' | 'success';
-        /** Boolean indicating if the message is in MD.  (Default is true) */
-        markdownMessage?: boolean;
-        /** Time in seconds to show the toast.  Set to 0 to disable automatic dismissal. (Default is 5) */
-        timeout?: number;
-        /** Boolean indicating if the close button should be shown. (Default is false) */
-        showCloseButton?: boolean;
-    } = {}) {
-        const messageParams: BaseToastMessageProps & typeof options = {
+    showToastPopup<
+        MessageType extends 'error' | 'info' | 'success',
```

Comment:
Should this extend ValidMessageType here?

Reply:
Ahh yeah, good catch